DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Centre/Administrator/AdministratorControllerTests.cs

```diff
@@ -188,5 +188,40 @@ public void UnlockAccount_unlocks_account_and_returns_to_page()
             A.CallTo(() => userDataService.UpdateAdminUserFailedLoginCount(1, 0)).MustHaveHappened();
             result.Should().BeRedirectToActionResult().WithActionName("Index");
         }
+
+        [Test]
+        public void DeactivateAdminUser_does_not_deactivate_admin_user_without_confirmation()
+        {
+            // Given
+            A.CallTo(() => userDataService.GetAdminUserById(A<int>._)).Returns(UserTestHelper.GetDefaultAdminUser());
+            var adminUser = UserTestHelper.GetDefaultAdminUser(active: true);
+            var deactivateViewModel = new DeactivateAdminViewModel(adminUser) { Confirm = false };
+            const string expectedErrorMessage = "You must confirm before deactivating this account.";
+
+            // When
+            var result = administratorController.DeactivateAdmin(1, deactivateViewModel);
+
+            // Then
+            result.Should().BeViewResult().WithDefaultViewName().ModelAs<DeactivateAdminViewModel>();
+            administratorController.ModelState[nameof(DeactivateAdminViewModel.Confirm)].Errors[0].ErrorMessage.Should()
+                .BeEquivalentTo(expectedErrorMessage);
+        }
+
+        [Test]
+        public void DeactivateAdminUser_deactivate_admin_user_with_confirmation()
+        {
+            // Given
+            A.CallTo(() => userDataService.GetAdminUserById(A<int>._)).Returns(UserTestHelper.GetDefaultAdminUser());
+            var adminUser = UserTestHelper.GetDefaultAdminUser(active: true);
+            var deactivateViewModel = new DeactivateAdminViewModel(adminUser) { Confirm = true };
+
+            // When
+            var result = administratorController.DeactivateAdmin(1, deactivateViewModel);
+            var deactivatedAdminUser = result.Should().BeViewResult().Model as AdminUser;
+
+            // Then
+            deactivatedAdminUser?.Active.Should().Be(false);
+            result.Should().BeViewResult().WithViewName("DeactivateAdminConfirmation");

```

Comment:
Likewise, was expecting to see something like:
`            A.CallTo(() => userDataService.DeactivateAdmin(1)).MustHaveHappened();
`

Reply:
Added condition as suggested